dev/reference/best-practices/pd-scheduling.md

```diff
@@ -0,0 +1,269 @@
+
+### The speed of taking nodes offline is slow
+
+This scenario requires examining the generation and execution of Operators through related metrics.
+
+When an Operator is successfully generated but processes slow, possible reasons are:
+
+- The schedule speed is limited by default. You can adjust `leader-schedule-limit` or `replica-schedule-limit` to a larger value. Similarly, `max-pending-peer-count` and `max-snapshot-count` can also be properly enlarged.
+- Other scheduling tasks are running concurrently and competing in the system. You can refer to the solution in [the previous section](#leaderregion-is-not-evenly-distributed).
+- When you take a single node offline, a number of Region Leaders to be operated are concentrated on the offline node (about 1/3 under the configuration of 3 replicas), so the speed is limited by the speed at which this single node generates Snapshots. You can speed it up by manually adding an `evict-leader-scheduler` to migrate Leaders.
+
+If the corresponding Operator fails to generate, possible reasons are:
+
+- The Operator is stopped, or `replica-schedule-limit` is set to 0.
+- There is no proper node to migrate Regions. For example, if the capacity of nodes that replace the nodes of same Label is larger than 80%, PD will stop scheduling to avoid running out of storage space. In such case, you need to add more nodes or delete some data to free space.
+
+### The speed of putting nodes online is slow
+
+Currently, to take nodes online is scheduled through balance region mechanism, so you can refer to [Leader/Region is not evenly distributed](#leaderregion-is-not-evenly-distributed) for troubleshooting.
+
+### Hot Regions are not evenly distributed
+
+Hot Regions scheduling generally has the following problems:
+
+- There is a majority of hot Regions, but the scheduling speed cannot keep up with them to redistribute hot Regions in time.
+
+    **Solution**: adjust `hot-region-schedule-limit` to a larger value, and reduce the limit quota of other schedulers to speed up hot Regions scheduling. Or you can adjust `hot-region-cache-hits-threshold` to a smaller value to make PD sensitive to traffic changes.
+
+- A single Region with extensive traffics. For example, to scan a small table extensively is required in the production environment, which can also be detected from PD metrics. Since a single hotspot cannot be resolved by redistributing, you need to manually add a `split-region` Operator to redistribute such a Region.
+
+- The load of some nodes is significantly higher than that of other nodes from TiKV-related metrics, which becomes the bottleneck of the whole system. Currently, PD counts hotspots through traffic analysis. So it is possible that PD fails to identify hotspots in certain scenarios. For example, some Regions have a large number of point-and-check requests, which are not significant in terms of traffic, but high QPS of which leads to bottlenecks in key modules.
```

Comment:
```suggestion
- The load of some nodes is significantly higher than that of other nodes from TiKV-related metrics, which becomes the bottleneck of the whole system. Currently, PD counts hotspots through traffic analysis only, so it is possible that PD fails to identify hotspots in certain scenarios. For example, when there are intensive point lookup requests for some Regions, it might not be obvious to detect in traffic, but still the high QPS might lead to bottlenecks in key modules.
```

Reply:
Try not start a sentence with words like "so", "and", etc.

Reply:
point lookup/query 点查